server/config/passport.js

```diff
@@ -1,13 +1,19 @@
-const DiscordStrategy = require("passport-discord").Strategy;
+const DiscordStrategy = require("passport-discord");
 const User = require("../models/User");
 
 module.exports = function (passport) {
   passport.serializeUser((user, done) => {
     done(null, user.id);
   });
 
-  passport.deserializeUser((id, done) => {
-    User.findById(id, (err, user) => done(err, user));
+  passport.deserializeUser(async (id, done) => {
+    try {
+      const user = await User.findById(id);
+      if (!user) throw new Error("User not found");
+      done(null, user);
+    } catch (err) {
+      done(err);
+    }
```

Comment:
what is this doing that the previous code isn't doing?

Reply:
Sometimes I get errors from this function that stops the server. I thought that it was because `User.findById` is asynchronous, but `(id, done) => {...` doesn't have `async`?

Reply:
Both functions basically are dependent on `done` being called, so regardless of whether it's async or sync, it "resolves" when `done` is called.  I think you could actually do something as simple as:

```js
passport.deserializeUser((id, done) => {
    User.findById(id, done);
})
```

Or just (for the same reason):

```js
passport.deserializeUser(User.findById)
```

This is much less readable though so the original version is probably preferred over these code-golfy ones 🙃

Reply:
I'll change the code back to what it was and will let you know when it occurs again. I think it was in postman, not in the browser.